src/fonts/castlefonts_font.inc

```diff
@@ -296,27 +314,66 @@ begin
     DrawableImage.SmoothScaling := GetSmoothScaling;
 end;
 
-function TCastleFont.GetSmoothScaling: boolean;
+function TCastleFont.GetSmoothScaling: Boolean;
 begin
   Assert(FFont <> nil); // should never be called otherwise
   Result := Size <> FFont.Size;
 end;
 
 procedure TCastleFont.PrepareResources;
+
+  procedure AttachDistanceFieldShader;
+  var
+    VS, FS: String;
+  begin
+    VS := 'attribute vec2 vertex;' + LineEnding +
+          'attribute vec2 tex_coord;' + LineEnding +
+          'uniform vec2 viewport_size;' + LineEnding +
+          'varying vec2 tex_coord_frag;' + LineEnding +
+          'void main(void)' + LineEnding +
+          '{' + LineEnding +
+          '  gl_Position = vec4(vertex * 2.0 / viewport_size - vec2(1.0), 0.0, 1.0);' + LineEnding +
+          '  tex_coord_frag = tex_coord;' + LineEnding +
+          '}' + LineEnding;
+    FS := 'varying vec2 tex_coord_frag;' + LineEnding +
+          'uniform sampler2D image_texture;' + LineEnding +
```

Comment:
Moreover, the shader code should be placed in a dedicated text file, processed just like existing `image.fs`.

Place it in 
- `src/base_rendering/glsl/source/distance_field_font.fs` 
- And for vertex shader, it seems you can just reuse it `src/base_rendering/glsl/source/image.vs` .

The subdirectory `base_rendering` is not the best for `fonts`, but this way the `distance_field_font.fs` will live alongside `image.fs` with which it should be synchronized.